Ginger/GingerCoreNET/Run/RunSetActions/RunSetActionHTMLReport.cs

```diff
@@ -111,41 +111,43 @@ public override void Execute(ReportInfo RI)
                     GingerRunner gr = new GingerRunner();
                     runSetFolder = gr.ExecutionLoggerManager.GetRunSetLastExecutionLogFolderOffline();                    
                 }
-                if (!string.IsNullOrEmpty(selectedHTMLReportTemplateID.ToString()))
+
+                string currentHTMLFolderName = string.Empty;
+                if (isHTMLReportFolderNameUsed && !String.IsNullOrEmpty(HTMLReportFolderName))
                 {
-                    ObservableList<HTMLReportConfiguration> HTMLReportConfigurations = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>();
-                    if (isHTMLReportFolderNameUsed && !String.IsNullOrEmpty(HTMLReportFolderName))
+                    if (!isHTMLReportPermanentFolderNameUsed)
                     {
-                        string currentHTMLFolderName = string.Empty;
-                        if (!isHTMLReportPermanentFolderNameUsed)
-                        {
-                            currentHTMLFolderName = HTMLReportFolderNameCalculated + "\\" + System.IO.Path.GetFileName(runSetFolder);
-                        }
-                        else
-                        {
-                            currentHTMLFolderName = HTMLReportFolderNameCalculated;
-                        }
-                        reportsResultFolder = Ginger.Reports.GingerExecutionReport.ExtensionMethods.CreateGingerExecutionReport(new ReportInfo(runSetFolder),
-                                                                                                                                false,
-                                                                                                                                HTMLReportConfigurations.Where(x => (x.ID == selectedHTMLReportTemplateID)).FirstOrDefault(),
-                                                                                                                                currentHTMLFolderName,
-                                                                                                                                isHTMLReportPermanentFolderNameUsed, currentConf.HTMLReportConfigurationMaximalFolderSize);
+                        currentHTMLFolderName = Path.Combine(HTMLReportFolderNameCalculated, System.IO.Path.GetFileName(runSetFolder));
                     }
                     else
                     {
-                        reportsResultFolder = Ginger.Reports.GingerExecutionReport.ExtensionMethods.CreateGingerExecutionReport(new ReportInfo(runSetFolder),
+                        currentHTMLFolderName = HTMLReportFolderNameCalculated;
+                    }
+                }
+                if (String.IsNullOrEmpty(currentHTMLFolderName))
```

Comment:
didn't really liked the IF statments here, it should have been with IF-Else statments, like IF Custome configured then use it Else If Artifact configured then use it Else use Solution default and then do actual Report.

and not check if Custom done and then check again if it done for artifact.

i'll merge it but please keep in mind for next devs.